sympy/solvers/tests/test_solvers.py

```diff
@@ -1420,14 +1420,22 @@ def test_issue_6792():
 
 def test_issues_6819_6820_6821_6248_8692():
     # issue 6821
+    from sympy.solvers.solveset import solveset
+    from sympy.sets import FiniteSet
     x, y = symbols('x y', real=True)
     assert solve(abs(x + 3) - 2*abs(x - 3)) == [1, 9]
     assert solve([abs(x) - 2, arg(x) - pi], x) == [(-2,), (2,)]
     assert set(solve(abs(x - 7) - 8)) == set([-S(1), S(15)])
 
     # issue 8692
-    assert solve(Eq(Abs(x + 1) + Abs(x**2 - 7), 9), x) == [
```

Comment:
This is expected. That is why in the test function the variables are defined as being real., 5 lines above "# issue 8692 ".